src/app/components/JumpTo/index.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  fireEvent,
+} from '../react-testing-library-with-providers';
+import JumpTo, { JumpToProps } from './index';
+import pidginArticleFixtureWithJumpToBlock from './fixtureData';
+import * as viewTracking from '../../hooks/useViewTracker';
+import * as clickTracking from '../../hooks/useClickTrackerHandler';
+
+describe('JumpTo Component', () => {
+  const jumpToBlock =
+    pidginArticleFixtureWithJumpToBlock.data.article.content.model.blocks.find(
+      block => block.type === 'jumpTo',
+    );
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const jumpToHeadings = jumpToBlock?.model.jumpToHeadings ?? [];
+
+  const defaultProps: JumpToProps = {
+    jumpToHeadings,
+    eventTrackingData: {
+      componentName: 'jumpto',
+    },
+  };
+
+  describe('Render jumpTo', () => {
+    it('renders the Jump To title', () => {
+      render(<JumpTo {...defaultProps} />);
+      const title = screen.getByText('Jump to');
+      expect(title).toBeInTheDocument();
+    });
+
+    it('renders the correct number of headings', () => {
+      render(<JumpTo {...defaultProps} />);
+      const headings = screen.getAllByRole('listitem');
+      expect(headings.length).toBe(jumpToHeadings.length);
+    });
+
+    it('renders each item with a link to the corresponding subheading on the same page', () => {
+      render(<JumpTo {...defaultProps} />);
+      const listItems = screen.getAllByRole('listitem');
+      listItems.forEach((item, index) => {
+        const link = item.querySelector('a');
+        const expectedHref = `#${jumpToHeadings[index].heading
+          .replace(/\s+/g, '-')
+          .replace(/'/g, '')}`;
+        expect(link).toHaveAttribute('href', expectedHref);
+      });
+    });
+  });
+
+  describe('Event Tracking', () => {
+    describe('View tracking', () => {
+      const viewTrackerSpy = jest.spyOn(viewTracking, 'default');
+
+      it('should not enable view tracking if event tracking data is not provided', () => {
+        render(<JumpTo jumpToHeadings={jumpToHeadings} />);
+
+        expect(viewTrackerSpy).toHaveBeenCalledWith(undefined);
```

Comment:
Is this test checking that the viewTracker IS called but with the value undefined? And is this what we expect or do we expect it to not be called at all if data is not provided? Same with the click tracker.

Reply:
See my comment below - is this line actually checking the set up of the tracker and not the event being fired?